cli/src/main/java/org/jboss/as/cli/handlers/trycatch/TryCatchFinallyControlFlow.java

```diff
@@ -111,6 +115,9 @@ void moveToCatch() throws CommandLineException {
     }
 
     void moveToFinally() throws CommandLineException {
+        if(finallyList == null) {
```

Comment:
You could move the list initialisation in the switch (IN_TRY and IN_CATCH). Doing so, no need to check for null and you will only initialise the list when needed.
Seems that you could get rid-off a call to break and merge the 2 cases.